spec/Candid.md

```diff
@@ -1229,7 +1229,7 @@ M(_ : reserved) = .
 M : <val> -> <constype> -> i8*
 M(null : opt <datatype>) = i8(0)
 M(?v   : opt <datatype>) = i8(1) M(v : <datatype>)
-M(v^N  : vec <datatype>) = leb128(N) M(v : <datatype>)^N
+M(v^N  : vec <datatype>) = leb128(N) M(v : <datatype>)^N  // N < 2097152 if <datatype> == null or reserved
```

Comment:
There are more types of size zero: records where all fields are zero. Maybe write 
> if `|M(v : <datatype>)| = 0`